specification/network/resource-manager/Microsoft.Network/stable/2024-01-01/virtualNetworkGateway.json

```diff
@@ -4480,6 +4492,39 @@
         "customBgpIpAddress"
       ],
       "description": "GatewayCustomBgpIpAddressIpConfiguration for a virtual network gateway connection."
+    },
+    "ConnectionAuthenticationType": {
+      "type": "string",
+      "description": "Gateway connection authentication type.",
+      "enum": [
+        "PSK",
+        "Certificate"
+      ],
+      "x-ms-enum": {
+        "name": "ConnectionAuthenticationType",
+        "modelAsString": true
+      }
+    },
+    "CertificateAuthentication": {
+      "type": "object",
+      "properties": {
+        "outboundAuthCertificate": {
+          "type": "string",
+          "description": "Outbound authentication certificate."
+        },
+        "inboundAuthCertificateSubjectName": {
+          "type": "string",
+          "description": "Inbound authentication certificate subject name."
+        },
+        "inboundAuthCertificateChain": {
```

Comment:
>inboundAuthCertificateChain [](http://example.com/codeflow?start=9&length=27)

Same here. If this array can contain secrets, it needs the x-ms-extension extension tag.